mizar/dp/mizar/workflows/builtins/pods/create.py

```diff
@@ -41,44 +42,42 @@ def requires(self):
         logger.info("Run {task}".format(task=self.__class__.__name__))
-        # name in self.param.body['metadata']['name']
-        # namespace in self.param.body['metadata']['namespace']
-        # label in self.param.body['metadata']['label']
-        # annotations in self.param.body['metadata']['annotations']
-
-        # logger.info("name: {}".format(self.param.body['metadata']['name']))
-        # logger.info("namespace: {}".format(
-        #     self.param.body['metadata']['namespace']))
-        # logger.info("label: {}".format(self.param.body['metadata']['labels']))
-        # logger.info("annotations: {}".format(
-        #     self.param.body['metadata']['annotations']))
-        # logger.info("hostNetwork: {}".format(
-        #     self.param.spec.get('hostNetwork', 'False')))
-        # logger.info("hostIP: {}".format(self.param.body['status']['hostIP']))
-        # logger.info("phase: {}".format(self.param.body['status']['phase']))
-        # logger.info("podIPs: {}".format(self.param.body['status']['podIPs']))
 
         spec = {
             'hostIP': self.param.body['status']['hostIP'],
             'name': self.param.body['metadata']['name'],
-            'namespace': self.param.body['metadata']['namespace'],
-            'tenant': '',
-            # TODO (Cathy) in case of arktos
-            # get VPC and net information from annotation
-            'vpc': OBJ_DEFAULTS.default_ep_vpc,
+            'type': 'k8s',
+            'namespace': self.param.body['metadata'].get('namespace', 'default'),
+            'tenant': self.param.body['metadata'].get('tenant', ''),
+            'vpc': self.param.body['metadata'].get('labels', {}).get(
+                    'arktos.futurewei.com/network', OBJ_DEFAULTS.default_ep_vpc),
             'net': OBJ_DEFAULTS.default_ep_net,
             'phase': self.param.body['status']['phase'],
-            # TODO (Cathy) in case of arktos get list of interfaces to create on
-            # the host (names)
+            'readiness': self.param.body['metadata'].get('annotations', {}).get(
+                    'arktos.futurewei.com/network-readiness', ''),
             'interfaces': [{'name': 'eth0'}]
         }
 
         logger.info("Pod spec {}".format(spec))
         spec['vni'] = vpc_opr.store_get(spec['vpc']).vni
         spec['droplet'] = droplet_opr.store_get_by_ip(spec['hostIP'])
 
-        # TODO (cathy): make sure not to trigger init or create simple endpoint
+        if 'arktos.futurewei.com/network' in self.param.body['metadata'].get('labels', {}):
+            spec['type'] = 'arktos'
+
+        if 'arktos.futurewei.com/nic' in self.param.body['metadata'].get('annotations', {}):
+            net_config = self.param.body['metadata']['annotations']['arktos.futurewei.com/nic']
+            configs = json.loads(net_config)
+            spec['interfaces'] = configs
+            for config in configs:
+                spec['net'] = config.get('subnet', OBJ_DEFAULTS.default_ep_net)
```

Comment:
I don't understand the use of this. If configs have multiple interfaces each with a separate network,  this loop overrides the network to be the last 'subnet' or default. I don't think that this is the expected behavior. Would you please explain?

Reply:
All the interfaces associated to a pod should be in the same subnet (correct me if I am wrong). Thus, we should only take one of the interface subnet that was passed in using yaml. 
I think the correct way should be: 
     if subnet in config: 
          spec['net'] = config['subnet']

Reply:
No, a POD can be multi-homed. Two interfaces on the pod may belong to different networks (and even different VPCs).